tests/integration/helpers.py

```diff
@@ -77,3 +77,30 @@ async def find_unit(ops_test: OpsTest, leader: bool, app=APP_NAME) -> ops.model.
             ret_unit = unit
 
     return ret_unit
+
+
+async def get_leader_id(ops_test: OpsTest) -> int:
+    """Returns the unit number of the juju leader unit."""
+    leader_unit_id = 0
+    for unit in ops_test.model.applications[APP_NAME].units:
+        if await unit.is_leader_from_status():
+            return leader_unit_id
+
+        leader_unit_id += 1
+
+    return leader_unit_id
```

Comment:
this function will work as you expect. IIRC ` ops_test.model.applications[APP_NAME].units` does not always return the units in order, ie I think it can return them in `unit/2`, `unit/0`, `unit/1`. (also in VM if you have 3 units, remove unit `unit/0` and then do `juju add-unit` you will then have `unit/1`, `unit/2`, `unit/3` 

instead I would parse the name with something like
```
        if await unit.is_leader_from_status():
            # parse unit id from unit name
            return = int(unit.name.split("/")[1])

Reply:
Done, thank you